File: language_benchmarks/xorshift32.cpp
#include <chrono>
#include <cstdint>
#include <iostream>

int main(int argc, char **argv) {
    if (argc != 3) return 2;
    const uint64_t iterations = std::stoull(argv[1]);
    uint32_t state = static_cast<uint32_t>(std::stoul(argv[2]));
    const auto started = std::chrono::steady_clock::now();
    for (uint64_t index = 0; index < iterations; ++index) {
        state ^= state << 13;
        state ^= state >> 17;
        state ^= state << 5;
    }
    const auto elapsed = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::steady_clock::now() - started).count();
    std::cout << "result=" << state << " loop_ns=" << elapsed << "\n";
    return 0;
}

File: language_benchmarks/xorshift32.js
if (process.argv.length !== 4) process.exit(2);
const iterations = Number.parseInt(process.argv[2], 10);
let state = Number.parseInt(process.argv[3], 10) >>> 0;
const started = process.hrtime.bigint();
for (let index = 0; index < iterations; index += 1) {
  state = (state ^ (state << 13)) >>> 0;
  state = (state ^ (state >>> 17)) >>> 0;
  state = (state ^ (state << 5)) >>> 0;
}
const elapsed = process.hrtime.bigint() - started;
console.log(`result=${state} loop_ns=${elapsed}`);
